kafka-handler/src/java/org/apache/hadoop/hive/kafka/KafkaSerDe.java

```diff
@@ -133,12 +134,40 @@
       Preconditions.checkArgument(!schemaFromProperty.isEmpty(), "Avro Schema is empty Can not go further");
       Schema schema = AvroSerdeUtils.getSchemaFor(schemaFromProperty);
       LOG.debug("Building Avro Reader with schema {}", schemaFromProperty);
-      bytesConverter = new AvroBytesConverter(schema);
+      bytesConverter = getByteConverterForAvroDelegate(schema, tbl);
     } else {
       bytesConverter = new BytesWritableConverter();
     }
   }
 
+  enum BytesConverterType {
+    CONFLUENT,
```

Comment:
Hi not sure, why we need this ? let say confluent changes the number of bytes to 4, then will have to add a new enum field ? I think adding skip bytes and document the confluent use case is better, What you think ?

Reply:
As long as we document it very well, I'm fine with that. I understand your thought process there, and it makes perfect sense. I'm also worried about developers who use Confluent not realizing that they need to skip bytes, so this is the value for them. It wouldn't be obvious if you are working with Confluent that you actually need to skip some bytes. Perhaps we can add some documentation on the Hive Confluence somewhere obvious stating this (not just as part of the documentation for this particular value).

Reply:
Confluent introducing a backwards incompatible change seems unlikely

Reply:
That makes sense, @cricket007. So far Confluent has done a good job preventing backward compatibility issues, so I don't see a reason to assume that they will do so in the future at the cost of a worse developer experience. I'd be fine leaving it as is.

Reply:
Overall, I'm somewhat in agreement with @b-slim here. There is little reason to make a specific "subtype" if it is just documented that `avro.skip.bytes=5` will get the necessary Avro payload. 

**However**, you would not know _which_ of those 5 bytes actually represents the schema ID in order to set `schema.literal` behind the scenes.